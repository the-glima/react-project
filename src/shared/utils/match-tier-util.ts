interface MatchTiers {
  tier: number
  info: {
    numbers: string
    euroNumbers: string
  }
}

export const matchTiers: MatchTiers[] = [
  {
    tier: 0,
    info: {
      numbers: `5 Numbers`,
      euroNumbers: `2 Euronumbers`
    }
  },
  {
    tier: 1,
    info: {
      numbers: `5 Numbers`,
      euroNumbers: `1 Euronumbers`
    }
  },
  {
    tier: 2,
    info: {
      numbers: `5 Numbers`,
      euroNumbers: `0 Euronumbers`
    }
  },
  {
    tier: 3,
    info: {
      numbers: `4 Numbers`,
      euroNumbers: `2 Euronumbers`
    }
  },
  {
    tier: 4,
    info: {
      numbers: `4 Numbers`,
      euroNumbers: `1 Euronumbers`
    }
  },
  {
    tier: 5,
    info: {
      numbers: `4 Numbers`,
      euroNumbers: `0 Euronumbers`
    }
  },
  {
    tier: 6,
    info: {
      numbers: `3 Numbers`,
      euroNumbers: `2 Euronumbers`
    }
  },
  {
    tier: 7,
    info: {
      numbers: `2 Numbers`,
      euroNumbers: `2 Euronumbers`
    }
  },
  {
    tier: 8,
    info: {
      numbers: `3 Numbers`,
      euroNumbers: `1 Euronumbers`
    }
  },
  {
    tier: 9,
    info: {
      numbers: `3 Numbers`,
      euroNumbers: `0 Euronumbers`
    }
  },
  {
    tier: 10,
    info: {
      numbers: `1 Numbers`,
      euroNumbers: `2 Euronumbers`
    }
  },
  {
    tier: 11,
    info: {
      numbers: `2 Numbers`,
      euroNumbers: `1 Euronumbers`
    }
  }
]

export const matchTierUtil = (index: number, collection = matchTiers) => {
  if (index > collection.length) return null

  const tier = collection.find((tier: MatchTiers) => tier.tier === index)

  return tier ? tier?.info : null
}
