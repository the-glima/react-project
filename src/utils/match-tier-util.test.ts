import {matchTierUtil, matchTiers} from './match-tier-util'

describe('Util: Match Tier', () => {
  it('should return the string number with thousand comma separator', () => {
    for (const [i] of matchTiers.entries()) {
      expect(matchTierUtil(i)).toMatchSnapshot()
    }
  })

  it('should return null if param is greater than matchTiers length', () => {
    expect(matchTierUtil(matchTiers.length + 1)).toBeNull()
  })
})
