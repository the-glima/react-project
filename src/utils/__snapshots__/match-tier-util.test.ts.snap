// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Util: Match Tier should return the string number with thousand comma separator 1`] = `
Object {
  "euroNumbers": "2 Euronumbers",
  "numbers": "5 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 2`] = `
Object {
  "euroNumbers": "1 Euronumbers",
  "numbers": "5 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 3`] = `
Object {
  "euroNumbers": "0 Euronumbers",
  "numbers": "5 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 4`] = `
Object {
  "euroNumbers": "2 Euronumbers",
  "numbers": "4 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 5`] = `
Object {
  "euroNumbers": "1 Euronumbers",
  "numbers": "4 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 6`] = `
Object {
  "euroNumbers": "0 Euronumbers",
  "numbers": "4 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 7`] = `
Object {
  "euroNumbers": "2 Euronumbers",
  "numbers": "3 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 8`] = `
Object {
  "euroNumbers": "2 Euronumbers",
  "numbers": "2 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 9`] = `
Object {
  "euroNumbers": "1 Euronumbers",
  "numbers": "3 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 10`] = `
Object {
  "euroNumbers": "0 Euronumbers",
  "numbers": "3 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 11`] = `
Object {
  "euroNumbers": "2 Euronumbers",
  "numbers": "1 Numbers",
}
`;

exports[`Util: Match Tier should return the string number with thousand comma separator 12`] = `
Object {
  "euroNumbers": "1 Euronumbers",
  "numbers": "2 Numbers",
}
`;
